src/util/threadnames.cpp

```diff
@@ -40,27 +41,31 @@ static void SetThreadName(const char* name)
 // global.
 #if defined(HAVE_THREAD_LOCAL)
 
-static thread_local std::string g_thread_name;
-const std::string& util::ThreadGetInternalName() { return g_thread_name; }
+static thread_local char g_thread_name[128]{'\0'};
```

Comment:
Seems like a weird comment to add without specfic/obvious motivation, and doesn't prevent the same thing from happening elsewhere in the codebase?

Reply:
Adding a comment would make sense. Storing a POD data structure in thread-local data instead of a C++ object that allocates on the heap is simpler and more widely supported.
Please also mention that this should be the only use of thread-local data in the program.

Reply:
Added a comment.